test/common/upstream/cds_api_impl_test.cc

```diff
@@ -93,6 +100,20 @@ class CdsApiImplTest : public testing::Test {
   Optional<envoy::api::v2::ConfigSource> eds_config_;
 };
 
+// Negative test for protoc-gen-validate constraints.
+TEST_F(CdsApiImplTest, ValidateFail) {
+  InSequence s;
+
+  setup(true);
+
+  Protobuf::RepeatedPtrField<envoy::api::v2::Cluster> clusters;
+  clusters.Add();
+
+  EXPECT_THROW(dynamic_cast<CdsApiImpl*>(cds_.get())->onConfigUpdate(clusters),
```

Comment:
EXPECT_THROW_WITH_MESSAGE? Maybe not convenient because exception contains entire debug string? Same elsewhere?

Reply:
It will get a bit fragile, since the current errors thrown by PGV are not very pretty, so will probably be evolved. This is why I added `ProtoValidationException` in this PR, so we can at least sure that these are PGV errors.

Reply:
OK sounds good.